app/views/pages/get-started/build-your-first-app/building-the-user-interface.liquid

```diff
@@ -0,0 +1,195 @@
+
+Liquid is parsed and executed **server-side**. All of the operations using Liquid are computed first on the server, and then the result of those operations is send to user browser.
+
+
+
+## Build the main application layout
+
+The main page’s HTML structure on platformOS is kept in `app/views/layouts/application.html.liquid`. Notice the file extension is `.liquid` – this is going to be a pattern for most of the files related to page’s structure. Specifically, the file named `application.html.liquid` is automatically used as a default structure for all pages in your application.
+
+Let’s start with a very simple HTML structure:
+
+#### app/views/layouts/application.html.liquid
+
+{% raw %}
+```liquid
+<!DOCTYPE html>
+<html>
+	<head>
+		<meta charset="utf-8">
+		<title>To do list</title>
+	</head>
+	<body>
+		<h1>My to do list</h1>
+
+		{{ content_for_layout }}
+	</body>
+</html>
+```
+{% endraw %}
+
+You may notice a special {% raw %}`{{ content_for_layout }}`{% endraw %} tag. This little thing here is Liquid.
+
+There will be many other tags and uses of Liquid in the future. This particular one is later automatically replaced by the actual page content that we will define in other `.liquid` files. So the content of other pages will be injected in place for {% raw %}`{{ content_for_layout }}`{% endraw %}. This makes the layout file a universal page structure for all of our content.
+
+{% include 'alert/tip', content: 'Remember that you need to [synchronise](/get-started/working-with-the-code-and-files/#upload-code-changes-automatically-when-you-save-the-file) or deploy your changes to the instance to see the results.' %}
+
+We can see this in action by creating another file – this one with the content that will be shown on the index page of our application.
+
+All of the pages (or _subpages_) of our web application needs to be placed in a separate folder: `app/views/pages/`. Start by creating the index file with a simple text:
+
+#### app/views/pages/index.liquid
+
+```html
+<h2>Hello world!</h2>
+```
+
+Now, when you go to your instance URL, you should see the application title that we defined in `application.html.liquid` as well as the content of `index.liquid`. Check out the source, it’s all there!
+
+
+## Add new page to your application
+
+Let’s say you would like to add an About me page to your application, that will show some personal information. We would like this page to be available under `/about` URL.
+
+On platformOS this is a super straight-forward operation. When you add a new file called `about.liquid` and save it in `app/view/pages/` folder, it will automatically be visible under /about URL on your instance. Basically – the file name becomes the URL.
+
+Ok, so let us actually do this. As described above – create a new file called `about.liquid` in `app/view/pages/` directory:
+
+#### app/views/pages/contact.liquid
+
+```html
+<h2>About me</h2>
+<p>
+	Hello! My name is [your name] and this is my first app built on platformOS
```

Comment:
I believe this would be a perfect example for showing named parameters, i.e. `slug: about/:name` and then rendering `Hello! My name is {{ context.params.name }}` , but could be too early to do it here :)

Reply:
I think it would be better to handle advanced routing as a separate article, or maybe extending the To do with the ability to see a task details that will use the routing with the task `id`. We'll see.